containers/unit_tests/TestUnorderedMap.hpp

```diff
@@ -293,8 +293,8 @@ void test_deep_copy(uint32_t num_nodes) {
   }
 }
 
-// FIXME_HIP wrong result in CI but works locally
-#ifndef KOKKOS_ENABLE_HIP
+// FIXME_HIP
+#if !defined(KOKKOS_ENABLE_HIP) || (HIP_VERSION > 400)
```

Comment:
I would prefer if HIP enabled and version less than 401

Reply:
But that would be wrong. I can do  `#if !defined(KOKKOS_ENABLE_HIP) || (defined(KOKKOS_ENABLE_HIP) && (HIP_VERSION > 400)` if you prefer.

Reply:
what's wrong with
```C++
#if !(defined(KOKKOS_ENABLE_HIP) && (HIP_VERSION < 401))

Reply:
ok I misunderstood what you wanted